src/gennylib/src/context.cpp

```diff
@@ -116,6 +124,8 @@ GlobalRateLimiter* WorkloadContext::getRateLimiter(const std::string& name, cons
         BOOST_THROW_EXCEPTION(
             std::logic_error("Cannot create rate-limiters after setup. Name tried: " + name));
     }
+
+    std::lock_guard<std::mutex> lk(_limiterLock);
```

Comment:
not that it matters a ton, but you can put these few items in a new scope so the `limiterLock` gets released sooner. I think it's only needed around the `if`:

```
{
  lock_guard lk{_limiter}
  if (_count(name) == 0) { _emplace.. }
}
```

Reply:
The rate-limiters are name-specific not thread-specific, so I think we need to lock while interacting with both the container and the limiters themselves further on.